### ISSUE TITLE
'Exception' with message 'Warning: Division by zero in vendor/magento/module-sales/Model/Order/Creditmemo/Total/Discount.php on line 32' in vendor/magento/framework/App/ErrorHandler.php:61

### ISSUE THREAD

**Issue by lsv** [magento/magento2#6731]
### Preconditions
1. [Magento 2.1.1](https://repo.magento.com/archives/magento/magento2-base/magento-magento2-base-2.1.1.0.zip)
### Steps to reproduce
1. Create a order with 0$ in shipment
2. Complete the order
3. Create new creditmemo
4. Add something in "Refund shipping"
### Expected result
1. I would suggest a error "You can not refund shipping when shipping is zero"

With the current error message the admin would try to create this creditmemo for long time, because the error is so "Try again later..."

Many errors around the site is just

```
try {
  ... massive code ...
} catch {
  ... Try again later ...
}
```
### Actual result
1. We can't save the credit memo right now.

**Comment by viktym** [comment]
Hi @lsv 
We've created internal ticket MAGETWO-59136 to address this issue.
Thanks for the feedback.

**Comment by magento-engcom-team** [comment]
@lsv, thank you for your report.
We've created internal ticket(s) MAGETWO-59136 to track progress on the issue.

**Comment by conflicker** [comment]
I got the same error. I created bundle product and add virtual products on it but the price of all child products is $0. When I tried to click add to cart I got error message:

> We can't add this item to your shopping cart right now.

Upon checking exception.log, I got the same error above.

**Comment by magento-engcom-team** [comment]
Hi @lsv

This ticket has been marked as `"Triage Wanted"` due to low user involvement over time. Over the next 2 weeks we are looking for additional community feedback to decide if it should be archived or not. More information on this is available on the GitHub [wiki](https://github.com/magento/magento2/wiki/Triage-Wanted).

Thank you for collaboration.

**Comment by lsv** [comment]
It can still be reproduced in master branch

**Comment by Rus0** [comment]
MM18MX

**Comment by magento-engcom-team** [comment]
Hi @Rus0. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [x] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [x] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by okorshenko** [comment]
Hi @lsv. Thank you for your report. 
The issue has been fixed in magento/magento2#18844 by @Rus0 in 2.3-develop branch
Related commit(s):
 - [dfa2f25534e05050fde25d2b4b960f5ecfa173bd](https://github.com/magento/magento2/commit/dfa2f25534e05050fde25d2b4b960f5ecfa173bd)
 - [4cf4884343010b2c0dc770ed3af6afb558b0d19b](https://github.com/magento/magento2/commit/4cf4884343010b2c0dc770ed3af6afb558b0d19b)
 - [a99c24fad345906e5cfc0bd4b79e60fcee92201b](https://github.com/magento/magento2/commit/a99c24fad345906e5cfc0bd4b79e60fcee92201b)
 - [b02dbc6f69ef9b2a48946218446302b204a6966e](https://github.com/magento/magento2/commit/b02dbc6f69ef9b2a48946218446302b204a6966e)
 - [d3e576f9ff770aa0d9475ff24f29eb0c4c8780b9](https://github.com/magento/magento2/commit/d3e576f9ff770aa0d9475ff24f29eb0c4c8780b9)
 - [02dc040145559d88a86a971b5c0e08076735534c](https://github.com/magento/magento2/commit/02dc040145559d88a86a971b5c0e08076735534c)
 - [73af54a06edfb2a2f4e2582f229dc4d301db9df4](https://github.com/magento/magento2/commit/73af54a06edfb2a2f4e2582f229dc4d301db9df4)

The fix will be available with the upcoming 2.3.1 release.

**Comment by ghost** [comment]
Hi @lsv , thank you for you report, this issue has already fixed in 2.3-develop branch, and will be available on 2.3.1 release.
